packages/sdk/src/node.ts

```diff
@@ -165,3 +184,47 @@ export function loadScriptNode(
     );
   });
 }
+
+async function loadModule(
+  url: string,
+  options: {
+    vm: any;
+    fetch: any;
+  },
+  parentContext?: any,
+) {
+  const { fetch, vm } = options;
+  const context =
+    parentContext ||
+    vm.createContext({
+      ...global,
+      Event,
+      URL,
+      URLSearchParams,
+      TextDecoder,
+      TextEncoder,
+      console,
+      require: eval('require'),
+      __dirname,
+      __filename,
+    });
+  const response = await fetch(url);
+  const code = await response.text();
+
+  const module: any = new vm.SourceTextModule(code, {
+    context,
```

Comment:
I do not think you should pass a custom VM context to the module. Attempting to copy the global context does not work for many cases, there are many more primitives than just 
```
  Event,
      URL,
      URLSearchParams,
      TextDecoder,
      TextEncoder,
      console,
      ```
      
      if you pass no context, then it will run in the current context and should operate like runInThisContext does for Script